File: jambeth/jAmbeth.Test/src/test/java/de/osthus/ambeth/transfer/ITestService.java
package de.osthus.ambeth.transfer;

import java.util.Date;
import java.util.List;

import de.osthus.ambeth.annotation.XmlType;
import de.osthus.ambeth.model.Material;
import de.osthus.ambeth.model.MaterialGroup;

@XmlType
public interface ITestService
{
	void noParamNoReturn();

	void primitiveParamNoReturn(int param);

	void dateParamNoReturn(Date param);

	void primitiveArrayParamNoReturn(int[] param);

	void primitiveListParamNoReturn(List<Integer> param);

	void entityParamNoReturn(MaterialGroup param);

	void entityWithRelationParamNoReturn(Material param);

	void mixedParamsNoReturn(int number, Material material1, String text, MaterialGroup materialGroup, Material material2, Date date);

	int noParamPrimitiveReturn();

	Date noParamDateReturn();

	int[] noParamPrimitiveArrayReturn();

	List<Integer> noParamPrimitiveListReturn();

	MaterialGroup noParamEntityReturn();

	Material noParamEntityWithRelationReturn();
}

File: ambeth/Ambeth.Test/ambeth/xml/oriwrapper/ITestService.cs
using System;
using System.Collections.Generic;
using De.Osthus.Ambeth.Annotation;
using De.Osthus.Ambeth.Test.Model;

namespace De.Osthus.Ambeth.Xml.Test.Transfer
{
    [XmlType(Name = "ITestService")]
    public interface ITestService
    {
        void NoParamNoReturn();

        void PrimitiveParamNoReturn(int param);

        void DateParamNoReturn(DateTime param);

        void PrimitiveArrayParamNoReturn(int[] param);

        void PrimitiveListParamNoReturn(List<Int32> param);

        void EntityParamNoReturn(MaterialGroup param);

        void EntityWithRelationParamNoReturn(Material param);

        void MixedParamsNoReturn(int number, Material material1, String text, MaterialGroup materialGroup, Material material2, DateTime date);

        int NoParamPrimitiveReturn();

        DateTime NoParamDateReturn();

        int[] NoParamPrimitiveArrayReturn();

        List<Int32> NoParamPrimitiveListReturn();

        MaterialGroup NoParamEntityReturn();

        Material NoParamEntityWithRelationReturn();
    }
}
